File: examples/weather/src/test/java/com/example/weather/TestCityProvider.java
/*
 * Copyright (c) 2026 Konstantin Pavlov and contributors.
 */

package com.example.weather;

import com.example.weather.spi.CityNotFoundException;
import com.example.weather.spi.CityProvider;
import com.example.weather.spi.WeatherObservation;
import com.example.weather.spi.WeatherProvider;

import java.util.List;
import java.util.concurrent.CompletableFuture;

final class TestCityProvider implements CityProvider {

    private static final List<String> KNOWN_CITIES = List.of("London", "Los Angeles", "Tokyo", "Tallinn");




    @Override
    public CompletableFuture<List<String>> searchCities(String query) {
        if (query.isBlank()) {
            return CompletableFuture.completedFuture(List.of());
        }
        var matches = KNOWN_CITIES.stream()
            .filter(city -> city.toLowerCase().startsWith(query.toLowerCase()))
            .toList();
        return CompletableFuture.completedFuture(matches);
    }
}

File: examples/weather-mcp-kotlin/src/test/kotlin/com/example/weather/TestCityProvider.kt
// Copyright (c) 2026 Konstantin Pavlov and contributors.

package com.example.weather

import com.example.weather.spi.CityProvider

class TestCityProvider : CityProvider {
    override fun searchCities(query: String): List<String> {
        if (query.isBlank()) return emptyList()
        return KNOWN_CITIES.filter { it.lowercase().startsWith(query.lowercase()) }
    }

    companion object {
        private val KNOWN_CITIES = listOf("London", "Los Angeles", "Tokyo", "Tallinn")
    }
}
